### ISSUE TITLE
Search.js example Bug results dont display when delete letter in word and retype

### ISSUE THREAD

**Issue by Jieiku** [Keats/gutenberg#1856]
The primary Zola Themes site has this bug, and therefore nearly every theme created also has this bug. (because they use it as an example according to docs: https://www.getzola.org/documentation/content/search/)

To reproduce:

- go here: https://www.getzola.org/themes/
- Search for the term zola (good, results appear)
- press backspace to delete the a from zola (good, results disappear)
- Now type a again to form the word zola (BUG, no results)

![Peek 2022-05-12 11-04](https://user-images.githubusercontent.com/106644/168140038-ade9532c-56e7-40a7-bd0c-3ad51066f94e.gif)

While designing I took the time to look at every single theme to not overlook any useful features, and one thing I noticed was that this bug did NOT exist for [adidoks](https://www.getzola.org/themes/adidoks/)

Also while not necessarily a bug the search feature seems incomplete to not show the full results when pressing the "Enter Key" or the search icon. The only theme that I seen that did this and had a search icon/button was [zolastrap](https://www.getzola.org/themes/zolastrap/)

What I have done is extended the search functionality, to combine what these two scripts do, for the code from zolastrap I really only used the construct (function names, form actions), I rewrote the code so that it does this:

- copy the contents of the suggestions popup box.
- rename the suggestions popup box identifier from `#suggestions` to `#results` to allow me to style it slightly different.
- add a header to the results page with the search term and a button to close the results.
- display the contents into the main tag of the page.

![Peek 2022-05-12 11-06](https://user-images.githubusercontent.com/106644/168140339-68e701c3-0935-468f-a640-671e1b8494fc.gif)

You can try it out here: https://abridge.netlify.app/

**Comment by Keats** [comment]
That seems like a fairly simple JS bug, I'll have a look

**Comment by Jieiku** [comment]
Thank you! The only search.js that I could find that did not have this bug was the the [Adidoks theme search.js](https://github.com/aaranxu/adidoks/blob/main/static/js/search.js)

**Comment by Jieiku** [comment]
Update, I have improved the [search.js used on Abridge Theme](https://github.com/Jieiku/abridge/blob/master/static/search.js). My first creation of this script used inline javascript on the form action (When you press enter or the search icon to go to a results Page). I now use the onload event of the window to set the forms action directly from the search.js code instead of inline on the forms action.

The result is that I was able to enable Content Security Policy without setting unsafe-inline for the script src.
